src/planscape/base/conditions_test.py

```diff
@@ -58,6 +58,13 @@ def test_average_propagates_other_nodata(self):
             self.assertTrue(np.all(np.nan_to_num(average)
                             == np.nan_to_num(expected)))
 
+    def test_nan_not_defined_as_nodata(self):
+        condition1 = np.array([[np.nan], [np.nan]])
+        condition2 = np.array([[np.nan], [np.nan]])
+
+        self.assertRaises(KeyError, average_condition,
+                          2, [condition1, condition2])
+
 
 class WeightedAverageTest(unittest.TestCase):
```

Comment:
Can we add tests for the new "convert_nodata_to_nan" function? two or three should be enough.